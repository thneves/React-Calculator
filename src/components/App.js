import React from 'react';
import '../index.css';
import Display from './Display';
import Panel from './Panel';
import '../styles/App.css';
import Calculate from '../logic/calculate';
import Button from './Button';

function App() {
  Calculate({ total: null, next: null, operation: null }, <Button />);
  return (
    <>
      <div className="calculator">
        <Display result="10" />
        <Panel />
      </div>
    </>
  );
}

export default App;
