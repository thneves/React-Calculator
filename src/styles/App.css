.calculator {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 400px;
  margin: 100px auto 0 auto;
}
