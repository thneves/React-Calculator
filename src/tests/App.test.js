import React from 'react';
import TestRenderer from 'react-test-renderer';
import App from '../components/App';

it('should render App component', () => {
  const app = TestRenderer
    .create(<App />)
    .toJSON();
  expect(app).toMatchSnapshot();
});
