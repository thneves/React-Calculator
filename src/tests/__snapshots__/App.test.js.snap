// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App component 1`] = `
<div
  className="back"
>
  <div
    className="calculator"
  >
    <div
      className="display"
    >
      <h2
        className="display-numbers"
      >
        0
      </h2>
    </div>
    <div
      className="panel"
    >
      <div
        className="panel-row"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          /
        </button>
      </div>
      <div
        className="panel-row"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="panel-row"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="panel-row"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="panel-row"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
